src/QueryReflection/QuerySimulation.php

```diff
@@ -72,13 +75,21 @@ public static function simulateParamValueType(Type $paramType, bool $preparedPar
         }
 
         $stringType = new StringType();
-        if ($stringType->isSuperTypeOf($paramType)->yes() || $paramType instanceof ObjectType && $paramType->isInstanceOf(Stringable::class)->yes()) {
+        $isStringableObjectType = $paramType instanceof ObjectType
+            && $paramType->isInstanceOf(Stringable::class)->yes();
+        $isDateTimeType = $paramType instanceof ObjectType
```

Comment:
I don't think the date-time case you added here works as it is right now.
but lets separate the mysqli refactoring beforehand and discuss this stuff when the PR itself gets smaller and better managable.